sensor/common/centralclient/client.go

```diff
@@ -213,25 +220,76 @@ func (c *Client) parseTLSChallengeResponse(challenge *v1.TLSChallengeResponse) (
 		return nil, errors.Wrap(err, "reading CA cert")
 	}
 
-	x509CertChain, err := x509utils.ParseCertificateChain(trustInfo.GetCertChain())
-	if err != nil {
-		return nil, errors.Wrap(err, "parsing Central cert chain")
+	// Try primary CA first
+	err = validateCertChain(trustInfo.GetCertChain(), challenge.TrustInfoSerialized,
+		challenge.Signature, rootCAs, "primary")
+	if err == nil {
+		return &trustInfo, nil
 	}
 
-	if len(x509CertChain) == 0 {
-		return nil, errors.New("parsing Central chain was empty, expected certificate chain")
+	// Central might be using a certificate chain based on the secondary CA certificate, let's check that as well
+	if len(trustInfo.GetSecondaryCertChain()) == 0 {
+		return nil, errors.Wrap(err, "validating primary Central certificate chain (no secondary certificate chain present)")
+	}
+	err = validateCertChain(trustInfo.GetSecondaryCertChain(), challenge.TrustInfoSerialized,
+		challenge.SignatureSecondaryCa, rootCAs, "secondary")
+	if err != nil {
+		return nil, err
 	}
 
-	err = verifyCentralCertificateChain(x509CertChain, rootCAs)
+	return &trustInfo, nil
+}
+
+func validateCertChain(chainBytes [][]byte, trustInfoSerialized []byte, signature []byte, rootCAs *x509.CertPool, caName string) error {
+	certChain, err := x509utils.ParseCertificateChain(chainBytes)
 	if err != nil {
-		return nil, errors.Wrap(err, "validating certificate chain")
+		return errors.Wrapf(err, "parsing %s Central certificate chain", caName)
+	}
+
+	if len(certChain) == 0 {
+		return fmt.Errorf("parsing %s Central certificate chain, expected non-empty certificate chain", caName)
+	}
+
+	if err := verifyCentralCertificateChain(certChain, rootCAs); err != nil {
+		return errors.Wrapf(err, "verifying %s Central certificate chain", caName)
+	}
+
+	if err := verifySignatureAgainstCertificate(certChain[0], trustInfoSerialized, signature); err != nil {
+		return errors.Wrapf(err, "verifying payload signature with %s CA", caName)
 	}
 
-	err = verifySignatureAgainstCertificate(x509CertChain[0], challenge.TrustInfoSerialized, challenge.Signature)
+	return nil
+}
+
+// extractCentralCAsFromTrustInfo assumes that trustInfo was already verified
+func extractCentralCAsFromTrustInfo(trustInfo *v1.TrustInfo) []*x509.Certificate {
+	var centralCAs []*x509.Certificate
+
+	if primaryCACert, err := getCACertificate(trustInfo.GetCertChain()); err == nil {
+		centralCAs = append(centralCAs, primaryCACert)
+	}
+
+	if len(trustInfo.GetSecondaryCertChain()) > 0 {
+		if secondaryCACert, err := getCACertificate(trustInfo.GetSecondaryCertChain()); err == nil {
+			centralCAs = append(centralCAs, secondaryCACert)
+		}
+	}
```

Comment:
nit, skips a getter call:

```suggestion
	if certChain := trustInfo.GetSecondaryCertChain(); len(certChain) > 0 {
		if secondaryCACert, err := getCACertificate(certChain); err == nil {
			centralCAs = append(centralCAs, secondaryCACert)
		}
	}
```

Feel free to close.